src/main/java/ch/njol/skript/structures/StructVariables.java

```diff
@@ -140,14 +140,24 @@ public boolean hasDefaultVariables() {
 		public List<NonNullPair<String, Object>> getVariables() {
 			return variables;
 		}
+		
+		private boolean isLoaded() {
+			return loaded;
+		}
 	}
 
 	@Override
 	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult, EntryContainer entryContainer) {
 		SectionNode node = entryContainer.getSource();
 		node.convertToEntries(0, "=");
-
-		List<NonNullPair<String, Object>> variables = new ArrayList<>();
+		List<NonNullPair<String, Object>> variables;
+		Script script = getParser().getCurrentScript();
```

Comment:
What happens if a user just writes down `variables:` in effect commands? does it throw a NPE?

Reply:
Nothing should happen since effect commands are parsed as an effect :)